javascript/fluent.js

```diff
@@ -97,17 +97,19 @@
 			}
 		});
 
-		$('.cms-panel-deferred').entwine({
-			/**
-			 * Ensure that any deferred panel URLs include the locale parameter in their URL
-			 */
-			onadd: function() {
-				var url = this.attr('data-url'),
-					newUrl = urlForLocale(url, $.cookie('FluentLocale_CMS'));
+        $('.cms-panel-deferred').entwine({
+            /**
+             * Ensure that any deferred panel URLs include the locale parameter in their URL
+             */
+            onadd: function () {
+                var url = this.attr('data-url'),
+                    newUrl = urlForLocale(url, $.cookie('FluentLocale_CMS')),
+                    // Ensure "incoming" data-url is properly filtered before being re-applied into the DOM
+                    i18nNewUrl = ss.i18n.sprintf(this.attr('data-url'), newUrl);
```

Comment:
So we build the url from data-url and locale, and then sprintf the result back into data-url again?